### ISSUE TITLE
Crash after upgrading to Geolocator 2.0.0+

### ISSUE THREAD

**Issue by mhirons** [jamesmontemagno/Xamarin.Plugins#179]
After upgrading to Geolocator 2.0.0 or higher from 1.1.1, I experience a crash on app startup.

I am using an Activity called SplashActivity to display a splash screen.  Then, in OnCreate, I just start the MainActivity.

The error below indicates it is something to do with my SplashActivity, but I am not sure how to fix this.  I copied SplashActivity below since it is so small.

Any ideas?

[ERROR] FATAL UNHANDLED EXCEPTION: System.NotSupportedException: Unable to activate instance of type ConnectMobileCapture.Droid.SplashActivity from native handle 0xca90001d (key_handle 0x412e03e0). ---> System.MissingMethodException: No constructor found for ConnectMobileCapture.Droid.SplashActivity::.ctor(System.IntPtr, Android.Runtime.JniHandleOwnership) ---> Java.Interop.JavaLocationException: Exception of type 'Java.Interop.JavaLocationException' was thrown.

SplashActivity.cs
```
namespace ConnectMobileCapture.Droid
{
    [Activity(Theme = "@style/Theme.Splash", //Indicates the theme to use for this activity
             MainLauncher = true, //Set it as boot activity
             ScreenOrientation = Android.Content.PM.ScreenOrientation.Portrait, // Start in portrait mode?
             ConfigurationChanges = Android.Content.PM.ConfigChanges.Orientation & Android.Content.PM.ConfigChanges.ScreenSize, // Indicate that we will handle orientation and screensize changes (by ignoring them)
             NoHistory = true)] //Doesn't place it in back stack
    public class SplashActivity : Activity
    {
        protected override void OnCreate(Bundle bundle)
        {
            base.OnCreate(bundle);
            //System.Threading.Thread.Sleep(3000); //Let's wait awhile...
            this.StartActivity(typeof(MainActivity));
            this.Dispose();
        }
    }
}
```

**Comment by jamesmontemagno** [comment]
Are you compiling against API 23?

**Comment by mhirons** [comment]
Yes, I am compiling against API 23.
I started looking through some more commit logs and I see a PermissionsReadme.txt that indicates you are required to add an override called OnRequestPermissionsAsyncResult().  Do I need to do that, or is that something just the Geolocator plugin needs?
Thanks,Matt

Date: Wed, 13 Jan 2016 10:13:19 -0800

**Comment by jamesmontemagno** [comment]
Yes, this needs to be integrated into the app. Take a look at the readme on Geolocator. It explains everything.

**Comment by mhirons** [comment]
I apologize, but I must be missing something.

The app was working with API 23 and Geolocator 1.1.1.  The only change was upgrading Geolocator.  I can roll back to 1.1.1 and it starts working again.

In the Permissions readme, I see that I need to add the OnRequestPermissionsResult() override method to each of my Actions.  I did that.

Other than that, I don't see anything that mentions changes that need to be made by me (other than adding ACCESS_COARSE_LOCATION & ACCESS_FINE_LOCATION, which I already had).  I see that it says it automatically adds needed android permissions and it checks for permissions before StartListening.  However, the app never gets to the point where it creates the Geo object.

I put a breakpoint on the first line of OnCreate in my MainActivity and it does not hit before the crash.  So, the crash is after I call StartActivity(typeof(MainActivity)) and before MainActivity OnCreate.

I will create a stripped down project tomorrow and try to narrow down what might be causing this.

Thanks for helping.

**Comment by jamesmontemagno** [comment]
The largest change is that I also have a dependency on the CurrentActivity plugin with is API 14+, as is the Permissions Plugin. You should see a MainApplication in there. A small repo would be nice that I can test though.

**Comment by mhirons** [comment]
OK, I have a little more information.  Hopefully this can help to narrow down the issue.

I created a new project with only the Geolocator plugin and it works as it should.

I created another project and mimicked my actual solution, this time adding a Splash Screen activity.  With the splash screen, it crashes.  So, it doesn't like something there.

I'm new to git, so hopefully this works. I created a test repo with the crashing code.
https://github.com/mhirons/TestCode
